common/src/main/java/com/wynntils/core/features/overlays/OverlayManager.java

```diff
@@ -212,15 +221,13 @@ public Set<Overlay> getOverlays() {
         return overlayInfoMap.keySet();
     }
 
-    public OverlayInfo getOverlayInfo(Overlay overlay) {
-        return overlayInfoMap.getOrDefault(overlay, null);
-    }
-
     public Feature getOverlayParent(Overlay overlay) {
         return overlayParent.get(overlay);
     }
 
     public boolean isEnabled(Overlay overlay) {
         return enabledOverlays.contains(overlay);
     }
+
+    private record OverlayRenderInfo(RenderEvent.ElementType elementType, RenderState renderState) {}
```

Comment:
This is like a super-class to OverlayGroup and OverlayInfo? Maybe we should express it as a such? Or annotations cannot have a hierarchy?

Reply:
This is a "data" class to hold info about overlay render info. Annotations cannot have custom class fields not have hierarchy. 

I just mentioned it in discord at the same time you reviewed this :) : `the overlay annotations are a problem too. I think I might remove those too`

Reply:
Ok. I can see the need for some refactoring here, but I guess it'll have to do for now.